aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -785,15 +792,48 @@ internal class RealAWSCognitoAuthPlugin(
         onSuccess: Consumer<AuthUpdateAttributeResult>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
```

Comment:
nit: I wonder why the parameters of `updateUserAttribute` and `updateUserAttributes` were created differently even though could have been just plural version of one another.

Reply:
Should we deprecate singular version? Let's discuss.